tests/diff/float_values_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diff
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func TestFloatValues(t *testing.T) {
+	t.Parallel()
+
+	ctx, db := setup(t)
+
+	t.Run("Insert", func(t *testing.T) {
+		t.Parallel()
+
+		for name, tc := range map[string]struct {
+			doc      bson.D
+			expected mongo.CommandError
+		}{
+			"NaN": {
+				doc: bson.D{{"_id", "1"}, {"foo", math.NaN()}},
+				expected: mongo.CommandError{
+					Code:    2,
+					Name:    "BadValue",
+					Message: `wire.OpMsg.Document: validation failed for { _id: "1", foo: nan.0 } with: NaN is not supported`,
```

Comment:
`nan.0`?

Reply:
This output is from [FormatAnyValue](https://github.com/FerretDB/FerretDB/blob/d3cbadfa98be1c035dd318dafa9e5525f2aa9203/internal/types/path.go#L237). This is how MongoDB format it.

Reply:
That's an interesting question: on the one hand, as we don't support `NaN`s anymore, we don't need to support that formatting, and we can offer something better than `nan.0` for `NaN`s. On the other hand, someone who has experience with MongoDB might expect to see `nan.0` in the response 🤔